app/src/test/kotlin/compiler/intermediate_form/FunctionsUtilitiesTest.kt

```diff
@@ -768,4 +768,80 @@ class FunctionDependenciesAnalyzerTest {
 
         assertContentEquals(expectedCallGraph, actualCallGraph)
     }
+
+    @Test fun `test comparison by reference - sets`() {
```

Comment:
I added one more test to my previous PR (I realized it is missing during coding the current task).